workspaces/ui-v2/src/spectacle-implementations/cloud-viewer.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from 'react';
+import { useRouteMatch, useParams, Switch } from 'react-router-dom';
+import { Provider as BaseUrlProvider } from '../optic-components/hooks/useBaseUrl';
+import { makeSpectacle } from '@useoptic/spectacle';
+import { useEffect, useState } from 'react';
+import { DocumentationPages } from '../optic-components/pages/docs/DocumentationPage';
+import { AsyncStatus, SpectacleStore } from './spectacle-provider';
+import { Loading } from '../optic-components/navigation/Loading';
+import { DiffReviewEnvironments } from '../optic-components/pages/diffs/ReviewDiffPages';
+import { InMemoryInteractionLoaderStore } from './interaction-loader';
+import { IBaseSpectacle, SpectacleInput } from '@useoptic/spectacle';
+import { IForkableSpectacle } from '@useoptic/spectacle';
+import { InMemoryOpticContextBuilder } from '@useoptic/spectacle/build/in-memory';
+import { CapturesServiceStore } from '../optic-components/hooks/useCapturesHook';
+import { IOpticContext } from '@useoptic/spectacle';
+import { ChangelogPages } from '../optic-components/pages/changelog/ChangelogPages';
+
+export default function CloudViewer() {
+  const match = useRouteMatch();
+  const params = useParams<{ specId: string }>();
+  const { specId } = params;
+  const task: InMemorySpectacleDependenciesLoader = async () => {
+    const loadExample = async () => {
+      const apiBase = process.env.REACT_APP_API_URL;
+      const response = await fetch(`${apiBase}/api/specs/${specId}`, {
+        headers: { accept: 'application/json' },
+      });
+      if (!response.ok) {
+        throw new Error(`could not find spec ${specId}`);
+      }
+      const responseJson = await response.json();
+      let signedUrl = responseJson.read_url;
+
+      if(!signedUrl){
+        throw new Error(`No read url found: ${JSON.stringify(responseJson)}`);
+      }
+
+      let contentReq = await fetch(signedUrl);
+      if(!contentReq.ok){
+        throw new Error(`Unable to fetch spec ${specId}`);
+      }
+
+      let spec = await contentReq.json();
+      return spec;
+    };
+    const [example, opticEngine] = await Promise.all([
+      loadExample(),
+      import('@useoptic/diff-engine-wasm/engine/browser'),
+    ]);
+    return {
+      events: example.events,
+      samples: example.session.samples,
```

Comment:
So.. per our convo @devdoshi, I tried this with `samples: []` since you said that the gitbot won't be providing samples, but the ui seems to be unhappy with that. Is this just a "limbo" thing, or is there something worth thinking about here?

Reply:
can you send me a screencast of the issue?